Avoid cryptic error message when attempting to create a non-namespaced handler/action

timothyfisherdev opened the pull request:
**Provide a narrative description of what you are trying to accomplish:**

Provide a less cryptic error message for users trying to create a handler and/or action when not typing in a namespaced class.

- [x] Is this related to quality assurance?
  <!-- Detail why the changes are necessary -->

When you install Zend Expressive and go to create your first action/handler with the CLI,

Entering something like:

`composer expressive action:create HomePageAction`

results in the following cryptic error message that is not so friendly:

`Fatal error: Uncaught TypeError: substr() expects parameter 3 to be integer, boolean given in [path/to/zendframework/zend-expressive-tooling/src/CreateHandler/TemplateResolutionTrait.php]:44`

The error occurs in line 44 because the handler/action specified is not namespaced, therefore it does not have any backslash characters in the string, making strpos return boolean false in the 3rd argument of substr.

Using a conditional return it will fallback to the CreateHandlerException and say:

`"Unable to match [HomePageAction] to an autoloadable PSR-4 namespace"`

This is the behavior that occurs if you do the same thing with the middleware:create command.

Just a friendlier message for beginners.

Commit:
avoid cryptic error message when attempting to create a non-namespaced handler/action

weierophinney commented:
This is great!

Can you please add a test demonstrating that the exception is thrown, instead of the fatal error?

Commit:
Add test

timothyfisherdev commented:
Turns out it was only happening when there was a template engine installed (when [251-260](https://github.com/zendframework/zend-expressive-tooling/blob/cb12b0f9e6cf53dddee228d744cc9f206476c02c/src/CreateHandler/CreateHandlerCommand.php#L251-L260) executed), so I updated the test accordingly and fixed one other method in the trait that I noticed from doing the test.

weierophinney commented:
Thanks, @timothyfisherdev! <br><br>:octocat: *Sent from [GH](http://nodegh.io).*
